### PR TITLE
feature: update plan and subscription responses

### PR DESCRIPTION
**Pull request by jackiemunroe**
* added plan and product metadata to plan and subscription responses to support the implementation of tiered plans
* extracted methods for plan and subscription formatting to make verifying response formatting easier
* updated and added tests

https://jira.mozilla.com/browse/COPS-860

### CODE REVIEW

File: src/sub/payments.py [hunk 1]
```diff
@@ -393,52 +397,21 @@ def create_return_data(subscriptions) -> JsonDict:
...
-                        "plan_id": subscription["plan"]["id"],
-                        "status": subscription["status"],
-                        "subscription_id": subscription["id"],
-                        "cancel_at_period_end": subscription["cancel_at_period_end"],
-                        "failure_code": intents["failure_code"],
-                        "failure_message": intents["failure_message"],
-                    }
+                    format_subscription(subscription, product, intents)
                 )
                 continue
```
Comment (marty331) [inline src/sub/payments.py]: continue is not required and can have undesired effects.

File: src/sub/payments.py [hunk 2]
```diff
@@ -537,33 +510,46 @@ def create_update_data(customer) -> Dict[str, Any]:
...
-                        "plan_id": subscription["plan"]["id"],
-                        "status": subscription["status"],
-                        "cancel_at_period_end": subscription["cancel_at_period_end"],
-                        "subscription_id": subscription["id"],
-                        "failure_code": intents["failure_code"],
-                        "failure_message": intents["failure_message"],
-                    }
+                    format_subscription(subscription, product, intents)
                 )
                 continue
```
Comment (marty331) [inline src/sub/payments.py]: same as above, continue is not required and can have unexpected effects.

File: src/sub/payments.py (hunk 1 above)
Comment (jackiemunroe) [reply src/sub/payments.py]: updated

File: src/sub/payments.py (hunk 2 above)
Comment (jackiemunroe) [reply src/sub/payments.py]: updated

**Pull request merged**